src/core/lib/surface/server.cc

```diff
@@ -1209,6 +1210,11 @@ void Server::ChannelData::InitTransport(RefCountedPtr<Server> server,
   grpc_transport_perform_op(transport, op);
 }
 
+bool Server::ChannelData::IsRegisteredMethod(const grpc_slice& host,
+                                             const grpc_slice& path) {
+  return GetRegisteredMethod(host, path) != nullptr;
```

Comment:
As we've discussed, the down-side of this is that we will wind up doing the registered method lookup twice for every server call, which seems sub-optimal for performance.  Can we have this code cache the result of `GetRegisteredMethod()` in a call context element somehow, so that the code in `Server::ChannelData::MakeCallPromise()` and `Server::CallData::StartNewRpc()` can just look at that value instead of calling `GetRegisteredMethod()` again?

Reply:
+1... we absolutely cannot do a second lookup here. Let's chat offline on how to do this.

Reply:
Yeah, adding it in another call context is relatively straightforward, but as you mentioned it would be a additional memory cost for all grpc calls, not just OTel enabled calls.

Reply:
Updated